lib/puppet/indirector/file_server.rb

```diff
@@ -29,6 +29,7 @@ def find(request)
     return nil unless path = mount.find(relative_path, request)
     result = model.new(path)
     result.links = request.options[:links] if request.options[:links]
+    result.checksum_type = request.options[:checksum_type] if request.options[:checksum_type]
```

Comment:
Also, seems like just passing the options hash along to the constructor of the model rather than setting each option explicitly should do the trick, although there's one method in `Puppet::FileServing::Fileset` we don't set in the indirector (`ignore=`).  Not sure if we want explicitness where we instantiate the model or just pass through the request options.